core-server/src/main/java/org/orbisgis/orbisserver/control/web/IndexController.java

```diff
@@ -61,46 +71,67 @@
 import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.io.DataInputStream;
+import java.io.File;
 import java.io.InputStream;
 import java.util.List;
+import java.util.ArrayList;
 
 /**
  * Very light instance of DefaultController containing a WpsServer.
  *
  * @author Sylvain PALOMINOS
+ * @author Guillaume MANDE
  */
 @Controller
 public class IndexController extends DefaultController {
 
-    /**
-     * Gets the instance of the WpsServer.
-     */
+    /** Gets the instance of the WpsServer. */
     private WpsServer wpsServer = WpsServerManager.getWpsServer();
 
+    /** List all of the OrbisWPS processes into a String which be displayed on the index page. */
+    private String processesList = "";
+
+    /**  Used to display the corresponding xml file to GetCapabilities method. */
+    public WPSCapabilitiesType wpsCapabilitiesType = null;
+
+    /** Used by methods like DescribeProcessRequest to get all the ProcessSummaryType */
+    public List<ProcessSummaryType> processSummaryTypeList = null;
+
+    /** List of all identifier of all the processes */
+    private List<CodeType> codeTypeList = null;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IndexController.class);
+    private static final I18n I18N = I18nFactory.getI18n(IndexController.class);
+
     /**
      * Injects a template named 'index'.
      */
     @View("index")
-    Template index;
+    public Template index;
```

Comment:
It's a good practice to have the tests in the same packages (but in the `test` folder) as the class tested. 
I.E. : if you have `/src/main/java/org/orbisgis/MyClass.java`, you should have `/src/main/test/java/org/orbisgis/MyClassTest.java`

For the access to the template for the test, maybe you could make it private and add a getter on that field.